test/test/helpers/use-sinon/index.js

```diff
@@ -43,15 +43,17 @@ export function useFakeTimers( now = 0, clockCallback = noop ) {
  *
  * @param  {Object|Function} config The configuration to use, or a callback that is invoked with the sandbox instance
  * @param  {Function} sandboxCallback A callback function that is invoked with the sandbox instance
+ * @returns { {sandbox: Object} } Object with .sandbox property that points to the current sandbox instance
  */
 export function useSandbox( config, sandboxCallback = noop ) {
+	const wrapper = {};
 	if ( isFunction( config ) && sandboxCallback === noop ) {
 		sandboxCallback = config;
 		config = undefined;
 	}
 
 	before( function() {
-		this.sandbox = sinon.sandbox.create( config );
+		wrapper.sandbox = this.sandbox = sinon.sandbox.create( config );
```

Comment:
copy by value? 
`before` and `after` might be called after the function returns in which case we'll return `undefined` or something out of sync with the actual sandbox.

We can return a getter function, but I thought it might be too confusing.